File: yuanrenxue_code/bs_2/topic_1/topic_1.py
import requests
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
from cryptography.hazmat.backends import default_backend
import hashlib
import time

from Crypto.Util.Padding import pad
from Crypto.Cipher import AES


def gets(d):
    e = int(time.time())
    g = '666yuanrenxue66'

    key = hashlib.md5(g.encode()).digest()  # 使用 MD5 生成 16 字节的密钥
    cipher = AES.new(key, AES.MODE_ECB)
    data = str(e) + str(d)
    padded_data = pad(data.encode(), AES.block_size)
    h = cipher.encrypt(padded_data)
    print(h.decode('utf-8'))
    md5 = hashlib.md5()
    md5.update(h)
    token = md5.hexdigest()

    j = {
        'page': str(d),
        'token': token,
        'now': e
    }
    return j



# for i in range(1,6):
#     print(requests.post(url='https://match2023.yuanrenxue.cn/api/match2023/1', data=gets(i)).text)
#     break

File: yuanrenxue_code/bs_2/topic_1/topic_1.js

call = function(d) {
    var e = Date['now']()
      , f = a('crypto-js')
      , g = '666yuanrenxue66'
      , h = f['AES']['encrypt'](e + String(d), g, {
        'mode': f['mode']['ECB'],
        'padding': f['pad']['Pkcs7']
    })
      , i = '/api/match2023/1'
      , j = {
        'page': String(d),
        'token': f['MD5'](h['toString']())['toString'](),
        'now': e
    };
    $['ajax']({
        'url': i,
        'dataType': 'json',
        'async': !![],
        'data': j,
        'type': 'POST',
        'beforeSend': function(k) {},
        'success': function(k) {
            k = k['data'];
            let l = '';
            $['each'](k, function(m, n) {
                l += '<td>' + n['value'] + '</td>';
            }),
            $('.number')['text']('')['append'](l),
            $('.page-message')['removeClass']('active'),
            $('.page-message')['eq'](parseInt(d) - 0x1)['addClass']('active');
        },
        'complete': function() {},
        'error': function() {
            failedAlert('请求失败了，生而为虫，我很抱歉~'),
            $('.page-message')['removeClass']('active'),
            $('.page-message')['eq'](0x0)['addClass']('active');
        }
    });
}
d = 1-5